sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JsonInferSchema.scala

```diff
@@ -115,13 +121,19 @@ private[sql] class JsonInferSchema(options: JSONOptions) extends Serializable {
         // record fields' types have been combined.
         NullType
 
-      case VALUE_STRING if options.prefersDecimal =>
+      case VALUE_STRING =>
+        val field = parser.getText
         val decimalTry = allCatch opt {
```

Comment:
Yea, I think this was a mistake. Previously if `prefersDecimal` was `false` (by default), it won't try decimal casting. Now looks we're trying decimal try always. @bersprockets, can you open a PR to fix it? I think we can just make it `lazy`.

Reply:
We shouldn't. The `opt` method calls body by name: `def opt[U >: T](body: => U): Option[U]` It should try infer if `options.prefersDecimal` is false.

Reply:
@HyukjinKwon What's the problem here. Could you give more context (JIRA, PR), please.

Reply:
Here, SPARK-26711. You're cc'ed there as well :).

Reply:
The problem here is decimal conversion looks always being tried even when `prefersDecimal` is false. Previously, it checked `prefersDecimal` first so decimal try wasn't made. This looks causing performance regression.

I mean

```scala
scala> if (prefersDecimal) allCatch opt { println("im expensive"); true }
res0: Any = ()

scala>  allCatch opt { println("im expensive"); true }
im expensive
res1: Option[Boolean] = Some(true)
```

Reply:
and making it lazy will save us by short circuiting. I wanted to open a PR right away but wanted to let him open since this is what I investigated at SPARK-26711.